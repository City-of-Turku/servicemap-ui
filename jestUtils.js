import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });