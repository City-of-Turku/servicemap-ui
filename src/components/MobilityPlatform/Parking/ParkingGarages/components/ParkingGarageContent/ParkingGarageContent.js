import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../../styled/styled';
import TextComponent from '../../../../TextComponent';

const ParkingGarageContent = ({ item }) => {
  const intl = useIntl();

  const renderText = (msgId, value) => (
    <StyledTextContainer>
      <Typography variant="body2">
        {value ? intl.formatMessage({ id: msgId }, { value }) : intl.formatMessage({ id: msgId })}
      </Typography>
    </StyledTextContainer>
  );

  const translations = {
    parkingCapacity: 'mobilityPlatform.content.parking.capacity',
    disabledParking: 'mobilityPlatform.content.parking.disabled.capacity',
    chargers: 'mobilityPlatform.content.parking.chargers.capacity',
  };

  const names = {
    fi: item.name_fi,
    en: item.name_en,
    sv: item.name_sv,
  };

  const addressObj = {
    fi: item.address_fi,
    en: item.address_en,
    sv: item.address_sv,
  };

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <TextComponent textObj={names} isTitle />
      </StyledHeaderContainer>
      <div>
        <TextComponent messageId="mobilityPlatform.content.address" textObj={addressObj} />
        {item?.extra?.parking_spaces ? renderText(translations.parkingCapacity, item?.extra?.parking_spaces) : null}
        {item?.extra?.disabled_spaces ? renderText(translations.disabledParking, item?.extra?.disabled_spaces) : null}
        {item?.extra?.charging_stations !== '' ? renderText(translations.chargers, item?.extra?.charging_stations) : null}
        {item?.extra?.services?.fi !== '' ? <TextComponent textObj={item.extra.services} /> : null}
      </div>
    </StyledContainer>
  );
};

ParkingGarageContent.propTypes = {
  item: PropTypes.shape({
    address_fi: PropTypes.string,
    address_en: PropTypes.string,
    address_sv: PropTypes.string,
    name_fi: PropTypes.string,
    name_en: PropTypes.string,
    name_sv: PropTypes.string,
    extra: PropTypes.shape({
      parking_spaces: PropTypes.number,
      disabled_spaces: PropTypes.number,
      charging_stations: PropTypes.string,
      services: PropTypes.shape({
        fi: PropTypes.string,
        en: PropTypes.string,
        sv: PropTypes.string,
      }),
    }),
  }),
};

ParkingGarageContent.defaultProps = {
  item: {},
};

export default ParkingGarageContent;
