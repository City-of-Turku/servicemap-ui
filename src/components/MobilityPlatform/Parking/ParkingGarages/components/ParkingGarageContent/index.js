import ParkingGarageContent from './ParkingGarageContent';

export default ParkingGarageContent;
