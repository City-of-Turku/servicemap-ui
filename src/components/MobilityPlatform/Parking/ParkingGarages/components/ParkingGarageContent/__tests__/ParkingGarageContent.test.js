// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../../i18n/fi';
import { initialState } from '../../../../../../../redux/reducers/user';
import ParkingGarageContent from '../index';

const mockProps = {
  item: {
    name_fi: 'Pysäköintihalli',
    address_fi: 'Testikatu 1',
    extra: {
      parking_spaces: 620,
      disabled_spaces: 2,
      charging_stations: '1 x Type 2 22 kW',
      services: {
        fi: 'Sähköauton latauspiste, kameravalvonta',
      },
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<ParkingGarageContent />', () => {
  it('does match snapshot', () => {
    const { container } = renderWithProviders(<ParkingGarageContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ParkingGarageContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.item.name_fi);
    expect(p[1].textContent).toContain(
      `${finnishTranslations['mobilityPlatform.content.address'].replace(
        '{value}',
        `${mockProps.item.address_fi}`,
      )}`,
    );
    expect(p[2].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.parking.capacity'].replace(
      '{value}',
      `${mockProps.item.extra.parking_spaces}`,
    )}`);
    expect(p[3].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.parking.disabled.capacity'].replace(
      '{value}',
      `${mockProps.item.extra.disabled_spaces}`,
    )}`);
    expect(p[4].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.parking.chargers.capacity'].replace(
      '{value}',
      `${mockProps.item.extra.charging_stations}`,
    )}`);
    expect(p[5].textContent).toContain(mockProps.item.extra.services.fi);
  });
});
