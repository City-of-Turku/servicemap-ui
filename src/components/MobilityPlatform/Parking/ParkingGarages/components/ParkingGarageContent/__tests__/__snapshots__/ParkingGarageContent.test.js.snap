// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkingGarageContent /> does match snapshot 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1dznbe2"
    >
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Pysäköintihalli
        </p>
      </div>
    </div>
    <div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Osoite: Testikatu 1
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Parkkipaikkojen määrä: 620
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Invapaikkojen määrä: 2
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Sähkölatauspaikat: 1 x Type 2 22 kW
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Sähköauton latauspiste, kameravalvonta
        </p>
      </div>
    </div>
  </div>
</div>
`;
